tests/test_performance.py

```diff
@@ -0,0 +1,140 @@
+    date_rng = pd.date_range(start=min(dates), end=max(dates), freq="H")
+    raw_data = {"dt": date_rng}
+    df = pd.DataFrame.from_dict(raw_data)
+    df = df.assign(
+        month=pd.Categorical(df.dt.dt.month, categories=list(range(1, 13))),
+        weekday=pd.Categorical(df.dt.dt.weekday, categories=list(range(7))),
+        hour=pd.Categorical(df.dt.dt.hour, categories=list(range(24))),
+    )
+    return df
+
+
+def create_poisson_zero_hmm(X, mu, xis, observed):
+    S = 2
+    z_tt = tt.stack([tt.dot(X, xis[..., s, :]) for s in range(S)], axis=1)
+    Gammas_tt = pm.Deterministic("Gamma", multilogit_inv(z_tt))
+    gamma_0_rv = pm.Dirichlet("gamma_0", np.ones((S,)))
+
+    if type(observed) == np.ndarray:
+        T = X.shape[0]
+    else:
+        T = X.get_value().shape[0]
+
+    V_rv = DiscreteMarkovChain("V_t", Gammas_tt, gamma_0_rv, shape=T)
+    if type(observed) == np.ndarray:
+        V_rv.tag.test_value = (observed > 0) * 1
+    else:
+        V_rv.tag.test_value = (observed.get_value() > 0) * 1
+    Y_rv = SwitchingProcess(
+        "Y_t",
+        [pm.Constant.dist(0), pm.Constant.dist(mu)],
+        V_rv,
+        observed=observed,
+    )
+    return Y_rv
+
+
+def assert_posterior_predictives(model, Y, X, test_X, test_y, xi_true):
+    number_of_draws = 50
+
+    with model:
+        steps = [
+            FFBSStep([model.V_t]),
+            pm.NUTS(
+                vars=[
+                    model.gamma_0,
+                    model.Gamma,
+                ],
+                target_accept=0.90,
+            ),
+        ]
+
+    with model:
+        posterior_trace = pm.sample(
+            draws=number_of_draws,
+            step=steps,
+            random_seed=100,
+            return_inferencedata=True,
+            chains=1,
+            cores=1,
+            tune=number_of_draws // 2,
+            progressbar=True,
+            idata_kwargs={"dims": {"Y_t": ["dt"]}},
+        )
+
+    # Update the shared variable values
+    Y.set_value(np.ones(test_X.shape[0]))
+    X.set_value(test_X)
+
+    model.V_t.distribution.shape = (test_X.shape[0],)
+
+    assert np.abs(posterior_trace.posterior.xis - xi_true).mean() < 0.5
+
+    trace = posterior_trace.posterior.drop_vars(["Gamma", 'V_t'])
+
+    with theano.config.change_flags(compute_test_value="off"):
+        adds_pois_ppc = pm.sample_posterior_predictive(
+            trace,
+            var_names=["V_t", "Y_t", 'Gamma'],
+            model=model
+        )
+
+    assert (adds_pois_ppc['V_t'] - (test_y == 1) * 1).mean() > 0.75
+
+
+@pytest.mark.slow
+def test_time_variant_model_base():
+    df = gen_toy_data()
+
+    formula_str = "1 + C(weekday)"
+    X_df = patsy.dmatrix(formula_str, df, return_type="dataframe").values
+
+    xi_shape = X_df.shape[1]
+
+    xi_0_true = np.array([2.0, -5.0, -3.0, 0.0, 0.0, -5.0, -5.0]).reshape(xi_shape, 1)
+    xi_1_true = np.array([-2.0, -1.0, 3.0, 0.0, 0.0, 5.0, 5.0]).reshape(xi_shape, 1)
```

Comment:
We may need to use the `xi` parameters from the time-varying example notebook&mdash;or ones as similarly extreme as those; otherwise, it may be fundamentally difficult to verify good out-of-sample state predictions.

Reply:
Yes, Those that the exact parameters in the notebooks.